kotlinx-coroutines-core/jvm/src/flow/internal/FlowExceptions.kt

```diff
@@ -2,6 +2,10 @@ package kotlinx.coroutines.flow.internal
 
 import kotlinx.coroutines.*
 
+/**
+ * Implementation note: `owner` is an internal marked that is used ONLY for identity checks by coroutines machinery,
+ * and it's never exposed, thus it's safe to have it both `@Transient` and non-nullable.
```

Comment:
It seems possible to maliciously deserialize and use an `AbortFlowException` that would fail in ways that legal instances wouldn't fail. I'm not sure why not just mark `owner` as nullable: I don't think anything breaks if we have `@JvmField @Transient val owner: Any? = owner` where a constructor only accepts a non-nullable version. Why not do that instead?

Reply:
I thought about that as well and decided not to do it: our general contract is that `owner` should be unique, which nullable type will immediately defeat -- and also will open the door *for us* to introduce new bugs (also, additionally cognitive load on handling nulls).

All of that, for the sake of the really malicious actor that probably can work around their ways anyway

Reply:
> also will open the door for us to introduce new bugs

I don't see the risk if

> a constructor only accepts a non-nullable version

Reply:
That's harded to follow in the code itself, the concept of "potential malicious actor" starts poisoning the signatures and makes readers/operators authors/etc. to read through that